docs/v1/P4Runtime-Spec.mdk

```diff
@@ -2044,13 +2044,35 @@ When writing the config via a `SetForwardingPipelineConfig` RPC, the cookie
 field is optional. For this reason, the actual value is wrapped in its own
 message to clearly identify cases where a cookie is not present.
 
-# General Principles for Message Formatting
+# General Principles for Message Formatting {#sec-message-formatting-principles}
 
-## Set / Unset Protobuf Field
+## Default-valued Fields {#sec-default-valued-fields}
+
+There is a subtle distinction between the treatment of default-valued scalar
+fields vs default-valued message fields in P4Runtime.
+
+### Set / Unset Scalar Fields
+
+In Protobuf version 3 (*proto3*), the default value of scalar fields is `0` for
+numeric types such as `int32`, and the empty string `""` for string types
+(`string` and `bytes`). An application, such as the P4Runtime client or
+server, is **unable to distinguish** between an unset scalar field and a scalar
+field set to its default value. Therefore, we usually reserve the default
+values 0 and "" of scalar fields to mean "unset".
+
+In particular, 0 is not a valid P4 object ID and it
+is an error to specify 0 for any P4 object ID in a non-read request towards the
+switch, such as in a `WriteRequest` or a `SetForwardingPipelineConfigRequest`.
```

Comment:
nit: I know you just moved that paragraph, but "server" would be a more generic & appropriate term than "switch"

Reply:
Good point. Done!